Add example of conditional JSX expressions (#1332)
withastro/docs: src/pages/en/core-concepts/astro-components.md

```
:::

> Astro can conditionally display HTML using JSX ternary expressions.
> 
> ```astro title="src/components/ConditionalHtml.astro" "item"
> ---
> const item = true;
> ---
> {item === true && (
>     <p>Show me!<p>
>   )
> }
> ```
> 
> If alternate content needs to be displayed when the condition is false then you can use a full if/then/else expression.
> 
> ```astro title="src/components/ConditionalHtml.astro" "item"
> ---
> const item = true;
> ---
> {item === true ? (
>     <p>Show me!<p>
>   ) : (
>     <p>Else show me if not true</p>
>   )
> }
> ```

Reviewer: ``````suggestion
Astro can conditionally display HTML using JSX logical operators and ternary expressions.

```astro title="src/components/ConditionalHtml.astro" "item"
---
const visible = true;
---
{visible && <p>Show me!<p>}

{visible ? <p>Show me!<p> : <p>Else show me!</p>}
```
``````
Thanks for adding these great patterns, @drewtownchi! Very useful (I use these myself!) and a valuable addition.

I think these can be simplified for Docs purposes, since we try to display concise examples that point our readers towards "Astro-y" patterns. I think that showing the ternary pattern, but not a full "if/else" example, is probably sufficient enough for users to realize that they CAN display conditionally like this, and if they happen to be more comfortable with a full if/else, or want to account for more complicated logic, they would probably infer that they can write it that way.

So my suggestion would be to model this more something like the above, and we'll get some feedback from others, too!
